server/app/assets/javascripts/admin_programs.ts

```diff
@@ -24,88 +24,124 @@ class AdminPrograms {
     )
   }
 
-  // When the common intake checkbox is selected,
-  // the following fields should be disabled:
-  // - program category checkboxes (disabled and unchecked)
-  // - application steps
-  // - long program description (only if northstar UI is enabled)
+  /**
+   * Attaches a change event listener to the common intake checkbox to manage
+   * the disabled state of related form elements. When the common intake
+   * checkbox is checked, the following fields are disabled:
+   * 1. Program categories
+   * 2. Long program description, if NorthStar UI is enabled
+   * 3. All application steps
+   *
+   * When the checkbox is unchecked, all these elements are re-enabled and
+   * required indicators are shown again.
+   */
   static attachCommonIntakeChangeListener() {
     addEventListenerToElements('#common-intake-checkbox', 'click', () => {
       const commonIntakeCheckbox = <HTMLInputElement>(
         document.querySelector('#common-intake-checkbox')
       )
 
-      const programCategoryCheckboxes = document.querySelectorAll(
-        '[id^="checkbox-category"]',
+      // Program categories
+      this.updateUSWDSCheckboxesDisabledState(
+        /* fieldSelectors= */ '[id^="checkbox-category"]',
+        /* shouldDisable= */ commonIntakeCheckbox.checked,
       )
-      programCategoryCheckboxes.forEach((checkbox) => {
-        const category = checkbox as HTMLInputElement
-        if (commonIntakeCheckbox.checked) {
-          category.disabled = true
-          category.checked = false
-        } else {
-          category.disabled = false
-        }
-      })
 
+      // Long program description
       const longDescription = document.getElementById(
         'program-display-description-textarea',
       ) as HTMLInputElement
       const northStarUiEnabled =
         longDescription.dataset.northstarEnabled === 'true'
-      this.maybeDisableField(
-        longDescription,
-        commonIntakeCheckbox.checked && northStarUiEnabled,
+      this.updateTextFieldDisableState(
+        /* fieldElement= */ longDescription,
+        /* shouldDisable= */ commonIntakeCheckbox.checked && northStarUiEnabled,
       )
 
-      const applicationStepTitles = document.querySelectorAll(
-        'input[id^="apply-step"]',
-      )
-      const applicationStepDescriptions = document.querySelectorAll(
-        'textarea[id^="apply-step"]',
+      // Application steps
+      this.updateTextFieldsDisabledState(
```

Comment:
suggestion (blocking): Is it possible to combine `updateTextFieldsDisabledState` and `updateTextFieldDisableState`? They are so similarly named that it tripped me up for a minute. Alternatively, would you update the name of one of them so it's more unique?

Reply:
I can see the confusion. I think it's valuable to have them separate, so the test is easier to read. Updated the name of them so they reflect the parameter type